### PR TITLE
Migrated to GO SDK V2, all tests OK

### PR DESCRIPTION
**Pull request by megaproaktiv**
go test ./...
ok  	github.com/aws/aws-secretsmanager-caching-go/scintegtests	6.690s 
ok  	github.com/aws/aws-secretsmanager-caching-go/secretcache	0.455s

*Issue #, if available: https://github.com/aws/aws-secretsmanager-caching-go/issues/39#issue-1624487266

*Description of changes: Migrated to GO SDK v2

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,8 @@ module github.com/aws/aws-secretsmanager-caching-go
 
 go 1.12
```
Comment (simonmarty) [inline go.mod:3]: According to the [public docs](https://aws.github.io/aws-sdk-go-v2/docs/migrating/) the Go SDK v2 requires a minimum version of Go 1.15. We should increment this value to Go 1.15 likewise.

**Review by simonmarty — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (megaproaktiv) [reply go.mod:3]: Agreed.

File: go.mod [hunk 2]
```diff
@@ -1,5 +1,9 @@
 module github.com/aws/aws-secretsmanager-caching-go
 
-go 1.12
+go 1.15
```
Comment (jordanfinners) [inline go.mod:3]: Thanks for this PR!
Wonder if its worth bumping to go 1.19 or 1.20?

Comment (megaproaktiv) [reply go.mod:3]: The go.mod for aws-sdk-go-v2 says go 1.15
Depends on how you see the go.mod version. I think it says "the min version you need to run this.
So I would vote for 1.15 and increase it as soon newer features like generics are used.

**Comment by ahummel25** [comment]
Hello, curious if there is any movement on this PR.

Thanks!

File: scintegtests/integration_test.go [hunk 3]
```diff
@@ -370,18 +372,22 @@ func integTest_getSecretStringWithTTL(t *testing.T, api secretsmanageriface.Secr
...
 	cache, _ := secretcache.New(
 		func(c *secretcache.Cache) { c.Client = api },
 	)
 
 	secretName := "NoSuchSecret"
 	_, err := cache.GetSecretString(secretName)
 
+	var cnfe *types.ResourceNotFoundException
+
+	// 	} else if awsErr, _ := err.(awserr.Error); awsErr.Code() != secretsmanager.ErrCodeResourceNotFoundException {
```
Comment (simonmarty) [inline scintegtests/integration_test.go:385]: Why was this commented out?

File: secretcache/cache.go [hunk 4]
```diff
@@ -18,17 +18,16 @@ package secretcache
...
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
 )
 
 // Cache client for AWS Secrets Manager secrets.
 type Cache struct {
 	lru *lruCache
 	CacheConfig
-	Client secretsmanageriface.SecretsManagerAPI
+	//Client *secretsmanager.Client
```
Comment (simonmarty) [inline secretcache/cache.go:29]: Idem, I assume this can be removed.

File: secretcache/cacheItem.go [hunk 5]
```diff
@@ -83,7 +81,8 @@ func (ci *secretCacheItem) executeRefresh(ctx context.Context) (*secretsmanager.
 		SecretId: &ci.secretId,
 	}
 
-	result, err := ci.client.DescribeSecretWithContext(ctx, input, request.WithAppendUserAgent(userAgent()))
+	result, err := ci.client.DescribeSecret(ctx, input)
+	// result, err := ci.cacheObject.client.DescribeSecret(ctx, input)
```
Comment (simonmarty) [inline secretcache/cacheItem.go:85]: Idem

**Review by simonmarty — CHANGES_REQUESTED** [review]

**Comment by adapptor-kurt** [comment]
Is this PR still alive?

**Comment by gerhardwagner** [comment]
would be great to get this merged

**Comment by gerhardwagner** [comment]
is there still hope ?

**Comment by wlewis4321** [comment]
Collab'd into #80 and merged in.

**Pull request closed without merging**